bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/AbstractInformationControlManager.java

```diff
@@ -1210,8 +1211,15 @@ protected void hideInformationControl() {
 		if (fInformationControl != null) {
 			storeInformationControlBounds();
 			fInformationControl.setVisible(false);
-			if (fInformationControlCloser != null)
+			if (fInformationControlCloser != null) {
 				fInformationControlCloser.stop();
+			}
+			if (fInformationControl instanceof AbstractInformationControl infoControl) {
+				Shell shell= infoControl.getShell();
+				if (!shell.isDisposed()) {
+					shell.setVisible(false);
```

Comment:
Now seeing this line added below `fInformationControl.setVisible(false);` I wonder **why** is the shell not hidden by `fInformationControl.setVisible(false);` by default?

Could you please check that, and also in different re-incarnations of `fInformationControl`?

Reply:
The Java Doc hoveer is a BrowserInformationControl. The set visible will lead to the following method call: org.eclipse.jface.internal.text.html.BrowserInformationControl.setVisible(boolean) ..

There it is checking shell.isVisible() .. which returns usually false, when eclipse is minimized and is not going to apply the visibility to the shell anymore.. But it will be visible if eclipse will come back (but the listeners are not registered)

Reply:
> The Java Doc hoveer is a BrowserInformationControl.

Note, we have multiple implementations around in SDK alone: 
![image](https://github.com/user-attachments/assets/ea1b7d43-26c0-49c4-866d-4eda40f1bdda)

Ideally we should check at least few of them how they work with this change

> There it is checking shell.isVisible() .. which returns usually false, when eclipse is minimized and is not going to apply the visibility to the shell anymore

Which brings a question: should we fix it **there** (`org.eclipse.jface.text.AbstractInformationControl.setVisible(boolean)` and others)?

If not, I would add a comment over the new block saying that for whatever reason known implementations of `IInformationControl.setVisible()` tend to omit `shell.setVisible()` call in some cases.

The change was added in 91e32933fa40aea0335d03bdebbbd304bfecebf5 without proper commit message.
I assume it was some optimization ( `shell.setVisible(true)` may do some more work as needed)?

The question if we should either remove this optimization completely or only have it for "true" case?

Reply:
Right.. Let me check that.. Seems more complicated than expected..

Reply:
There are 6 Implementations which override the setVisible() Method of the AbstractInformationControl class. 4 of them are using the super call so the adjustment of the AbstractInformationControl class is enough for those. The other 2 have a similar check so the same adjustment needs to be done there locally.

BrowserInformationControl, XbaseInformationControl.

But to be totally sure, I tried to avoid unnecessary behavior changes when visibility is going to be changed to false, it will be set to false (without checking isVisible() ) .. But for the true case there is no change.. (to avoid flickering or some other stuff which I couldn't foresee) ..